test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -387,6 +391,13 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 				mu.Unlock()
 			}
 
+			if !concreteOp.SkipWaitToCompletion {
```

Comment:
This line could move  up to @Line380.

Instead of those check, why not resetting the registry only when collecting the metrics as my proposal here?
https://github.com/kubernetes/kubernetes/pull/97757/files

Reply:
> This line could move up to @Line380.

Nope, you have to wait for collectors to finish their collecting.

> why not resetting the registry only when collecting the metrics as my proposal here?

Upon completion of an operator (step) of each workload, we need to reset if needed - think about the case of InitPods.

Reply:
> Upon completion of an operator (step) of each workload, we need to reset if needed - think about the case of InitPods.

`InitPods` is a different case, iiuc, `collectMetrics` is set only with `measurePods`, we can clear all the stale metric before we are really measure something, all the data generated before could be discarded.

Reply:
I don't think it's a good idea to clear the metrics generated in **previous** steps at the beginning of the **current** step. Each test/step is better designed atomic and stateless - in other words, a test/step should clear its metrics upon its completion, instead of leaving the leftover for subsequent test/step to clear.

Reply:
while both approaches are probably functionally similar, I agree with @Huang-Wei that each step should be cleaning up after itself